revolutionpi/revolution_pi.go

```diff
@@ -145,13 +149,40 @@ func (b *revolutionPiBoard) DoCommand(ctx context.Context,
 ) (map[string]interface{}, error) {
 	resp := make(map[string]interface{})
 
-	_, ok := req["getStatus"]
-	if ok {
-		err := b.controlChip.showDeviceList()
+	pinMessage, exists := req[readAddressKey]
+	if exists {
+		pinName, ok := pinMessage.(string)
+		if !ok {
+			return nil, fmt.Errorf("error performing readAddress: expected string got %v", pinMessage)
+		}
+		pin := SPIVariable{strVarName: char32(pinName)}
+		err := b.controlChip.mapNameToAddress(&pin)
 		if err != nil {
 			return nil, err
 		}
-		resp["getStatus"] = "status ok"
+		b.controlChip.logger.Debugf("pin: %#v", pin)
+		switch pin.i16uLength {
+		case 1:
+			value, err := b.controlChip.getBitValue(int64(pin.i16uAddress), pin.i8uBit)
+			if err != nil {
+				return nil, err
+			}
+			resp[pinName] = value
+		default:
+			value := make([]byte, pin.i16uLength/8)
+			n, err := b.controlChip.fileHandle.ReadAt(value, int64(pin.i16uAddress))
+			if err != nil {
+				return nil, err
+			}
+			b.controlChip.logger.Debugf("Read %#d bytes", n)
+			resp[pinName], err = readFromBuffer(value, n)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+	if !exists {
+		return nil, fmt.Errorf("no valid commands found, got %#v", req)
```

Comment:
I had it setup like this because there were multiple DoCommands implemented before, where this check would validate that none of the commands were triggered. Going to leave it as is in case we decide to add more